services-camel/src/test/java/edu/unc/lib/dl/services/camel/longleaf/DeregisterLongleafRouteTest.java

```diff
@@ -0,0 +1,162 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.services.camel.longleaf;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.builder.NotifyBuilder;
+import org.jgroups.util.UUID;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.ContextHierarchy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import edu.unc.lib.dl.services.MessageSender;
+
+/**
+ * @author bbpennel
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextHierarchy({
+    @ContextConfiguration("/spring-test/jms-context.xml"),
+    @ContextConfiguration("/deregister-longleaf-router-context.xml")
+})
+public class DeregisterLongleafRouteTest {
+
+    @Autowired
+    private MessageSender messageSender;
+
+    @Autowired
+    private DeregisterLongleafProcessor processor;
+
+    @Autowired
+    private CamelContext cdrLongleaf;
+
+    @Rule
+    public final TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    private String longleafScript;
+    private String outputPath;
+    private List<String> output;
+
+    @Before
+    public void setup() throws Exception {
+        tmpFolder.create();
+        outputPath = tmpFolder.newFile().getPath();
+        longleafScript = LongleafTestHelpers.getLongleafScript(outputPath);
+        processor.setLongleafBaseCommand(longleafScript);
+    }
+
+    @Test
+    public void deregisterSingleBinary() throws Exception {
+        NotifyBuilder notify = new NotifyBuilder(cdrLongleaf)
+                .whenCompleted(2)
+                .create();
+
+        String contentUri = generateContentUri();
+        sendMessages(contentUri);
+
+        boolean result1 = notify.matches(5l, TimeUnit.SECONDS);
+        assertTrue("Deregister route not satisfied", result1);
+
+        output = LongleafTestHelpers.readOutput(outputPath);
+        assertDeregisterCalled(1);
+        assertDeregisterPaths(contentUri);
+    }
+
+    @Test
+    public void deregisterSingleBatch() throws Exception {
+        NotifyBuilder notify = new NotifyBuilder(cdrLongleaf)
+                .whenCompleted(1 + 3)
```

Comment:
Why the 1 + 3?

Reply:
I wrote it that was since there are two types of messages being completed, 1 batch message and 3 individual messages, since the notifyBuilder tracks the whole camel context rather than a specific route within it. Perhaps a comment would be better though